module_8/frontend/frontend_shop.conf

```diff
@@ -0,0 +1,49 @@
+upstream nest_server {
+  server localhost:3000; 
+}
+
+server {
+  listen 8080; 
+  server_name react-shop.com;
+
+  root /var/www/frontend/app;
+  index index.html;
+
+  location / {
+	try_files $uri $uri/ /index.html =404;
+  }
+
+  location /api/ {
+	proxy_pass http://nest_server/;
+
+	proxy_http_version 1.1;
+	proxy_set_header Host $host;
+	proxy_set_header X-NginX-Proxy true;
+	proxy_set_header X-Real-IP $remote_addr;
+	proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+	proxy_set_header X-Forwarded-Proto $scheme;
+  }
+}
+
+server {
+	listen 443 ssl http2;
+	listen [::]:443 ssl http2;
+	server_name react-shop.com; 
+
+	ssl_certificate /etc/ssl/certs/nginx.crt;
+	ssl_certificate_key /etc/ssl/private/nginx.key;
+
+	location / {
+		proxy_pass http://127.0.0.1:8080/; 
+	}
+}
+
+server {
+  listen 80;
+  listen [::]:80;
+  server_name react-shop.com; 
+
+  location / {

```

Comment:
FYI: you can add this rule for all requests that your server receives without any location:

```
server {
  listen 80;
  server_name react-shop.com;
  return 301 https://$host$request_uri;
}
```

https://www.digitalocean.com/community/tutorials/nginx-rewrite-url-rules

Reply:
fixed